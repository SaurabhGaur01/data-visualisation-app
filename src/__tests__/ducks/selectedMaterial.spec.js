import {
    setSelectedMaterial,
    selectedMaterialReducer,
} from '../../ducks/selectedMaterial';

describe('selectedMaterial Duck Tests', () => {
    const mockKey = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setSelectedMaterial should match the snapshot' , () => {
            expect(setSelectedMaterial(mockKey)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(selectedMaterialReducer()).toEqual('');
        });
        it('should return the passed state if called with no action' , () => {
            expect(selectedMaterialReducer(mockKey)).toEqual(mockKey);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(selectedMaterialReducer(mockKey, unknownAction)).toEqual(mockKey);
        });
        it('should return the value which is passed' , () => {
            expect(selectedMaterialReducer('', setSelectedMaterial(mockKey))).toEqual(mockKey);
        });
    });
});