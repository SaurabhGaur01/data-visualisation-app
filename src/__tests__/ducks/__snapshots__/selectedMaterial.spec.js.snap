// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectedMaterial Duck Tests Action Generator setSelectedMaterial should match the snapshot 1`] = `
Object {
  "type": "modules/data-visualisation-app/SET_SELECTED_MATERIAL",
  "value": "mockValue",
}
`;
